### DISCUSSION TITLE
Can't install SixLabors.Fonts v 2.0.0 into VS project targeting  .NetFramework version 4.8

### DISCUSSION THREAD

**Discussion by MikeReeder** [SixLabors/Fonts#361] [General]
In VS 2022 im trying to get all of my NuGet packages up to date. It fails on SixLabors.Fonts with the following message:

Could not install package 'SixLabors.Fonts 2.0.0'. You are trying to install this package into a project that targets '.NETFramework,Version=v4.8', but the package does not contain any assembly references or content files that are compatible with that framework. For more information, contact the package author

Any help would be appreciated!!

**Comment by tocsoft** [comment]
SixLabors.Fonts v2 targets .net 6.0+ you will have to stick with the v1 versions when targeting 4.8.

**Reply by MikeReeder** [reply]
Thanks very much tocsoft.

**Reply by tonyqus** [reply]
I'm trying to fix https://github.com/nissl-lab/npoi/issues/1630. However, I notice that SixLabors.Fonts 2.x only supports .NET 6. But NPOI supports .NET standard 2.0, .NET standard 2.1 and .NET 4.7.2.


Is there any reason for SixLabors.Fonts 2.x to only support .NET 6?

**Reply by JimBobSquarePants** [reply]
SixLabors.Fonts v2 targets .NET 6+ by design.

**Why we don’t support .NET Framework or netstandard in v2:**

- We rely on modern runtime guarantees for performance and correctness, including trimming/AOT, improved globalization, and APIs that simply don’t exist or behave consistently on older TFMs.  
- Carrying legacy targets slows us down and forces us into lowest-common-denominator constraints.  
- Open source libraries really should not be expected to indefinitely support .NET Framework. It is end-of-life and holding back progress.

There are probably quite a few issues with our V1 version that have since been fixed in V2 and I intend on doing more work soon to fix further V2 issues and add features in preparation for V3. I would recommend either multi targeting or working towards removing legacy target frameworks in your own software. It’s only going to get more difficult.
